onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,99 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xReturned == pdFAIL){
+        return 0;
+    }
+
+    
+    
+    if(xLightServiceQueue == NULL){
+	    
+	xLightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_SiZE, LIGHT_EVENT_SIZE);
+	    
+	if(xLightServiceQueue == NULL){
+        	return 0;
+	}
+    }
     /* USER CODE END */
     return 1;
 }
 
+uint16_t getLightSensorData(void)
+{
+
+	adcData_t lightData;
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &lightData);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return lightData.value;
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    light_event_t lightEvent = NULL_EVENT;
+    while(1){
+        if (xQueueReceive(xLightServiceQueue, &lightEvent, portMAX_DELAY) == pdTRUE) {
+		
+            if (lightEvent == MEASURE_LIGHT){
+            
+            	uint16_t lightSensorData = getLightSensorData();
+		const uint8_t dataSize = 6;
```

Comment:
I was wrong when I said the length should be 6. Good thing the compiler caught me and gave a warning:
```
onboarding/source/amb_light_service.c: In function 'lightServiceTask':
onboarding/source/amb_light_service.c:95:45: warning: '
   ' directive output may be truncated writing 2 bytes into a region of size between 1 and 5 [-Wformat-truncation=]
   95 |                 snprintf(data, dataSize, "%u\r\n", lightSensorData);
      |                                             ^~~~
onboarding/source/amb_light_service.c:95:17: note: 'snprintf' output between 4 and 8 bytes into a destination of size 6
   95 |                 snprintf(data, dataSize, "%u\r\n", lightSensorData);
```

`dataSize` should actually be 8. A `uint16_t` has a max value of 65535 which is 5 digits (also 5 characters). The `\r\n` is 2 more characters. And then all strings have a null character, `\0`, at the end so 1 more character. `5+2+1=8`, so the `data` array needs to have a size of at least 8.

No need to make this change tho. Just remember to look through all the warnings that the compiler gives when you try to build. It's pretty good at catching some common errors.

Other than that, this PR lgtm. Great job completing the challenge!